File: src/main/java/com/health/myapplication/callback/ItemTouchHelperCallback.java
package com.health.myapplication.callback;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.ItemTouchHelper;
import androidx.recyclerview.widget.RecyclerView;

public class ItemTouchHelperCallback extends ItemTouchHelper.Callback {
    private final ItemMoveListener itemMoveListener;
    public ItemTouchHelperCallback(ItemMoveListener listener){
        itemMoveListener=listener;
    }

    @Override
    public int getMovementFlags(@NonNull RecyclerView recyclerView, @NonNull RecyclerView.ViewHolder viewHolder) {
        int dragFlag = ItemTouchHelper.UP | ItemTouchHelper.DOWN;
        int swipeFlags = ItemTouchHelper.START | ItemTouchHelper.END;
        return makeMovementFlags(dragFlag, swipeFlags);
     //   return makeFlag( ItemTouchHelper.ACTION_STATE_DRAG , dragFlag);
    }

    @Override
    public boolean onMove(@NonNull RecyclerView recyclerView, @NonNull RecyclerView.ViewHolder viewHolder, @NonNull RecyclerView.ViewHolder target) {
        itemMoveListener.onItemMove(viewHolder.getAdapterPosition(), target.getAdapterPosition());
        return true;
    }

    @Override
    public void onSwiped(@NonNull RecyclerView.ViewHolder viewHolder, int direction) {
        itemMoveListener.onItemSwiped(viewHolder.itemView.getId());
    }

    public interface ItemMoveListener{
        void onItemMove(int fromPosition, int toPosition);
        void onItemSwiped(int id);
    }

}

File: src/main/java/com/health/myapplication/callback/ItemTouchHelperCallback.kt
package com.health.myapplication.callback

import androidx.recyclerview.widget.ItemTouchHelper
import androidx.recyclerview.widget.RecyclerView

class ItemTouchHelperCallback(private val itemMoveListener: ItemMoveListener) : ItemTouchHelper.Callback() {
    var dragFrom = -1
    var dragTo = -1
    override fun getMovementFlags(recyclerView: RecyclerView, viewHolder: RecyclerView.ViewHolder): Int {
        val dragFlag = ItemTouchHelper.UP or ItemTouchHelper.DOWN
        val swipeFlags = ItemTouchHelper.START or ItemTouchHelper.END
        return makeMovementFlags(dragFlag, swipeFlags)
        //   return makeFlag( ItemTouchHelper.ACTION_STATE_DRAG , dragFlag);
    }

    override fun onMove(recyclerView: RecyclerView, fromViewHolder: RecyclerView.ViewHolder, toViewHolder: RecyclerView.ViewHolder): Boolean {
        itemMoveListener.onItemMove(fromViewHolder.adapterPosition, toViewHolder.adapterPosition)
        return true
    }

    override fun onSwiped(viewHolder: RecyclerView.ViewHolder, direction: Int) {
        itemMoveListener.onItemSwiped(viewHolder.itemView.id)
    }
    override fun canDropOver(recyclerView: RecyclerView, current: RecyclerView.ViewHolder, target: RecyclerView.ViewHolder): Boolean {
        return current.itemViewType == target.itemViewType
    }
    override fun clearView(recyclerView: RecyclerView, viewHolder: RecyclerView.ViewHolder) {
        super.clearView(recyclerView, viewHolder)
        //if (dragFrom != -1 && dragTo != -1 && dragFrom != dragTo) {
            itemMoveListener.onItemMoveFinished()
       // }
       // dragTo = -1
       // dragFrom = -1
    }

    interface ItemMoveListener {
        fun onItemMove(fromPosition: Int, toPosition: Int)
        fun onItemSwiped(id: Int)
        fun onItemMoveFinished()
    }
}